lib/ProductOpener/Display.pm

```diff
@@ -1116,55 +1118,92 @@ sub display_list_of_tags($$) {
 			];
 	}	
 	
-	eval {
-		$log->debug("Executing MongoDB aggregate query", { query => $aggregate_parameters }) if $log->is_debug();
-		$results = $products_collection->aggregate( $aggregate_parameters );
-	};
-	if ($@) {
-		$log->warn("MongoDB error - retrying once", { error => $@ }) if $log->is_warn();
-		# maybe $connection auto-reconnects but $database and $products_collection still reference the old connection?
-		
-		# opening new connection
+	my $mongodb_query_ref = $aggregate_parameters;
+	
+	my $key = $server_domain . "/" . freeze($mongodb_query_ref);
+	
+	$log->debug("MongoDB aggregate query key", { "key" => $key}) if $log->is_debug();
```

Comment:
`{ "key" => $key}` should probably be `{ key => $key }` (also in following logging statements)